geodiff/src/drivers/sqliteutils.cpp

```diff
@@ -538,6 +570,26 @@ std::vector<std::string> sqliteColumnNames(
   return az;
 }
 
+std::string sqliteErrorMessage( sqlite3 *db, const std::string &description )
+{
+  std::string errMsg = std::string( sqlite3_errmsg( db ) );
+  std::string errCode = std::to_string( sqlite3_extended_errcode( db ) );
+  return description + " (SQLITE3 error [" + errCode + "]: " + errMsg + ")";
+}
+
+void logSqliteError( std::shared_ptr<Sqlite3Db> db, const Logger &logger, const std::string &description )
+{
+  std::string errMsg = db ? sqliteErrorMessage( db->get(), description ) : description + "(unknown SQLite error)";
+  logger.error( errMsg );
+}
+
+void throwSqliteError( sqlite3 *db, const Logger &logger, const std::string &description )
+{
+  std::string errMsg = db ? sqliteErrorMessage( db, description ) : description + "(unknown SQLite error)";
+  logger.error( errMsg );
```

Comment:
you do not need to pass logger here to this function. if you throw error, it is always somewhere catched and logged upsteam! (last instance is API calls in geodiff.cpp). If you remove logger here from the function, it can clean-up a bunch of calls upstream (since you may not need to pass logger to caller functions too)